packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/clientcore/ClientCoreClientMethodMapper.java

```diff
@@ -0,0 +1,1698 @@
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ClientModelProperty;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ClientModels;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ExternalDocumentation;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.GenericType;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.IType;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ImplementationDetails;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ListType;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.MethodPageDetails;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.MethodParameter;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.MethodPollingDetails;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ModelPropertySegment;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ParameterTransformations;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.PrimitiveType;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ProxyMethod;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ProxyMethodParameter;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ReturnValue;
+import com.microsoft.typespec.http.client.generator.core.model.javamodel.JavaVisibility;
+import com.microsoft.typespec.http.client.generator.core.util.ClientModelUtil;
+import com.microsoft.typespec.http.client.generator.core.util.MethodNamer;
+import com.microsoft.typespec.http.client.generator.core.util.MethodUtil;
+import com.microsoft.typespec.http.client.generator.core.util.ReturnTypeJavaDocAssembler;
+import com.microsoft.typespec.http.client.generator.core.util.SchemaUtil;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ClientCoreClientMethodMapper extends ClientMethodMapper {
+
+    private static final ClientCoreClientMethodMapper INSTANCE = new ClientCoreClientMethodMapper();
+
+    private ClientCoreClientMethodMapper() {
+    }
+
+    public static ClientCoreClientMethodMapper getInstance() {
+        return INSTANCE;
+    }
+
+    private static final Pattern ANYTHING_THEN_PERIOD = Pattern.compile(".*\\.");
+
+    private final Map<CacheKey, List<ClientMethod>> parsed = new ConcurrentHashMap<>();
+
+    private static class CacheKey {
```

Comment:
I don't think we need to duplicate these.

Also, this file seems pretty big, and potentially have lots of duplication with `ClientMethodMapper`. @anuchandy